### PR TITLE
Update Words for v3

### PR DESCRIPTION
**Pull request by Joseph-Edwards**
This PR updates `Words.hpp` for v3.

### CODE REVIEW

**Comment by Joseph-Edwards** [comment]
The API has now been streamlined so there is only one class for converting words to strings (both single words, and ranges of words), and similarly for strings to words. The remaining decisions that needs to be made for this PR are whether or not we want to combine `ToWord` and `ToString` into one class template, and `WordRange` and `StringRange` into one class template.

File: src/words.cpp [hunk 1]
```diff
@@ -368,18 +263,18 @@ namespace libsemigroups {
     return it->second;
   }
 
-  void to_word(std::string_view s, word_type& w) {
-    w.resize(s.size(), 0);
-    std::transform(s.cbegin(), s.cend(), w.begin(), [](char c) {
-      return human_readable_index(c);
-    });
-  }
+  // void to_word(std::string_view s, word_type& w) {
```
Comment (james-d-mitchell) [inline src/words.cpp:266]: Delete

**Comment by Joseph-Edwards** [comment]
CI will fail until the corresponding pybind11 PR is merged, but otherwise I believe this is good to go.

File: docs/DoxygenLayout.xml [hunk 2]
```diff
@@ -77,62 +77,84 @@
...
-    <tab type="user" visible="yes" url="@ref freeband_group" title="Radoszewski-Rytter"/>
+    <tab type="user" visible="yes" url="@ref freeband_group" title="Radoszewski-Rytter" />
     <tab type="user" visible="yes" url="_HEADING_" title="Bibliography" />
     <tab type="user" visible="yes" url="@ref citelist" title="Bibliography" />
     <tab type="user" visible="yes" url="_HEADING_" title="Further info" />
-    <tab type="user" url="https://github.com/libsemigroups/libsemigroups" title="GitHub"/>
+    <
+    <
+    <
+    <
```
Comment (james-d-mitchell) [inline docs/DoxygenLayout.xml:150]: Something has gone wrong here

File: include/libsemigroups/knuth-bendix.hpp [hunk 3]
```diff
@@ -578,7 +578,8 @@ namespace libsemigroups {
     //!
     //! \sa Presentation::validate_word.
     void validate_word(word_type const& w) const override {
-      std::string s = to_string(presentation(), w);
+      ToString    to_string(presentation().alphabet());
+      std::string s = to_string(w);
       return presentation().validate_word(s.cbegin(), s.cend());
```
Comment (james-d-mitchell) [inline include/libsemigroups/knuth-bendix.hpp:583]: I'm guessing this doesn't return anything, maybe remove the return here?

File: include/libsemigroups/knuth-bendix.tpp [hunk 4]
```diff
@@ -776,6 +778,10 @@ namespace libsemigroups {
   template <typename Rewriter, typename ReductionOrder>
   typename detail::internal_char_type
   KnuthBendix<Rewriter, ReductionOrder>::uint_to_internal_char(size_t a) {
+    // TODO What is this check for?
```
Comment (james-d-mitchell) [inline include/libsemigroups/knuth-bendix.tpp:781]: The check is just to ensure that the input value doesn't overflow the internal char type, seems legit to me

**Review by james-d-mitchell — COMMENTED** [review]
Couple of minor comments can't view the large diffs in words.*pp on my phone will do that later

**Review by james-d-mitchell — APPROVED** [review]

**Pull request merged**